<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GMDebuggerQtClass</class>
 <widget class="QMainWindow" name="GMDebuggerQtClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>762</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GMDebuggerQt</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_4">
    <item>
     <widget class="CodeEditor" name="scriptEdit">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>18</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>true</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionConnect"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>16</width>
     <height>16</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen_Script"/>
   <addaction name="separator"/>
   <addaction name="actionConnect"/>
   <addaction name="actionDisconnect"/>
   <addaction name="separator"/>
   <addaction name="actionStep_In"/>
   <addaction name="actionStep_Out"/>
   <addaction name="actionStep_Over"/>
   <addaction name="actionRun_Thread"/>
   <addaction name="actionStop_Thread"/>
   <addaction name="actionStop_All_Threads"/>
   <addaction name="actionRun_All_Threads"/>
   <addaction name="actionSet_Break_Point"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QDockWidget" name="dockThreads">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Threads</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QTableWidget" name="threadTable">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="showDropIndicator" stdset="0">
        <bool>false</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::NoDragDrop</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="sortingEnabled">
        <bool>true</bool>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
       <property name="columnCount">
        <number>4</number>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>75</number>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>50</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="verticalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderVisible">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>50</number>
       </attribute>
       <attribute name="verticalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>75</number>
       </attribute>
       <column>
        <property name="text">
         <string>Thread</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Status</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Function</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Script</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockCallstack">
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Callstack</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QTableWidget" name="callStack">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DropOnly</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="columnCount">
        <number>1</number>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>200</number>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>200</number>
       </attribute>
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>200</number>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>200</number>
       </attribute>
       <column>
        <property name="text">
         <string>Function</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockContext">
   <property name="windowTitle">
    <string>Locals</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTableWidget" name="context">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DropOnly</enum>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="columnCount">
        <number>3</number>
       </property>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>50</number>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>50</number>
       </attribute>
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>20</number>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>50</number>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>50</number>
       </attribute>
       <column>
        <property name="text">
         <string>Name</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Type</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Value</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockGlobals">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="windowTitle">
    <string>Globals</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QTreeWidget" name="globalsTable">
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <attribute name="headerDefaultSectionSize">
        <number>100</number>
       </attribute>
       <attribute name="headerMinimumSectionSize">
        <number>75</number>
       </attribute>
       <attribute name="headerDefaultSectionSize">
        <number>100</number>
       </attribute>
       <attribute name="headerMinimumSectionSize">
        <number>75</number>
       </attribute>
       <column>
        <property name="text">
         <string>Name</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Type</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Value</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockOutput">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QTextEdit" name="outputWindow">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionConnect">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/connect.png</normaloff>:/GMDebuggerQt/Resources/connect.png</iconset>
   </property>
   <property name="text">
    <string>Connect</string>
   </property>
   <property name="iconText">
    <string>&amp;Connect</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionStep_In">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/step_in.png</normaloff>:/GMDebuggerQt/Resources/step_in.png</iconset>
   </property>
   <property name="text">
    <string>Step In</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionStep_Out">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/step_out.png</normaloff>:/GMDebuggerQt/Resources/step_out.png</iconset>
   </property>
   <property name="text">
    <string>Step Out</string>
   </property>
   <property name="shortcut">
    <string>Shift+F11</string>
   </property>
  </action>
  <action name="actionStep_Over">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/step_over.png</normaloff>:/GMDebuggerQt/Resources/step_over.png</iconset>
   </property>
   <property name="text">
    <string>Step Over</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionRun_Thread">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/thread_run.png</normaloff>:/GMDebuggerQt/Resources/thread_run.png</iconset>
   </property>
   <property name="text">
    <string>Run Thread</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionStop_Thread">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/thread_stop.png</normaloff>:/GMDebuggerQt/Resources/thread_stop.png</iconset>
   </property>
   <property name="text">
    <string>Stop Thread</string>
   </property>
  </action>
  <action name="actionStop_All_Threads">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/thread_stopall.png</normaloff>:/GMDebuggerQt/Resources/thread_stopall.png</iconset>
   </property>
   <property name="text">
    <string>Stop All Threads</string>
   </property>
   <property name="shortcut">
    <string>Pause</string>
   </property>
  </action>
  <action name="actionRun_All_Threads">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/thread_runall.png</normaloff>:/GMDebuggerQt/Resources/thread_runall.png</iconset>
   </property>
   <property name="text">
    <string>Run All Threads</string>
   </property>
  </action>
  <action name="actionSet_Break_Point">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/breakpoint.png</normaloff>:/GMDebuggerQt/Resources/breakpoint.png</iconset>
   </property>
   <property name="text">
    <string>Set Break Point</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionDisconnect">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/disconnect.png</normaloff>:/GMDebuggerQt/Resources/disconnect.png</iconset>
   </property>
   <property name="text">
    <string>Disconnect</string>
   </property>
  </action>
  <action name="actionOpen_Script">
   <property name="icon">
    <iconset resource="gmdebuggerqt.qrc">
     <normaloff>:/GMDebuggerQt/Resources/script_open.png</normaloff>:/GMDebuggerQt/Resources/script_open.png</iconset>
   </property>
   <property name="text">
    <string>Open Script</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QTextEdit</extends>
   <header>codeeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="gmdebuggerqt.qrc"/>
 </resources>
 <connections/>
</ui>
